spec/requests/api/authentication_spec.rb

```diff
@@ -292,4 +292,62 @@ def systoken(server_guid, userid, timestamp)
       expect_result_to_have_keys(ENTRYPOINT_KEYS)
     end
   end
+
+  context "Role Based Authorization" do
+    before do
+      FactoryGirl.create(:vm_vmware, :name => "vm1")
+    end
+
+    def vms_get_test(new_identifier)
+      config = Api::ApiConfig.collections.vms.collection_actions.get.first
+      saved_identifier = config.identifier
+      config.identifier = new_identifier
+      yield
+      config.identifier = saved_identifier
```

Comment:
This should be in an ensure

```ruby
def vms_get_test
   # stuff
   yield
ensure
   config.identifier = saved_indentifier
end
```

This was if a test raises an exception, you can be sure it puts the settings back.

Alternately, the test should not modify settings in this way.  Instead, there should be probably be a stub_settings type method, like we do for other Settings.  stubbing ensures the values return to normal after test completion.